File: src/main/java/com/rewards_service/domain/model/Wallet.java
package com.rewards_service.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Wallet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    private UserAccount userAccount; // <-- change here

    private long points;

    private double balance; // money credited from redemptions

    // Convenience constructor
    public Wallet(UserAccount userAccount) {
        this.userAccount = userAccount;
        this.points = 0L;
        this.balance = 0.0;
    }
}

File: src/main/kotlin/com/rewards_service/domain/model/Wallet.kt
package com.rewards_service.domain.model

import jakarta.persistence.*

@Entity
class Wallet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @OneToOne
    var userAccount: UserAccount? = null

    var points: Long = 0L

    var balance: Double = 0.0

    constructor()

    constructor(userAccount: UserAccount) {
        this.userAccount = userAccount
        this.points = 0L
        this.balance = 0.0
    }
}
